### DISCUSSION TITLE
Add `less` as a dependency to `circumflex`

### DISCUSSION THREAD

**Discussion by bensadeh** [Homebrew/discussions#3794] [Tap maintenance and Homebrew development] [answered]
Hi,

I am the author for [`circumflex`](https://github.com/bensadeh/circumflex) and I was wondering if it would be okay to change the `circumflex` [formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/circumflex.rb) to depend on `less` without any qualifications. Would it be okay if I opened up a PR to change this line:

```ruby
# Requires less 576 or later for --use-color
uses_from_macos "less", since: :monterey
``` 
to
```ruby
depends_on "less"
```

The reason for the change is that I am depending on the latest version of `less` and users reported errors using the formula and not having less being installed automatically as a dependency, see [here](https://news.ycombinator.com/item?id=33200091) for example.

**Comment by SMillerDev** [comment, marked as answer]
Yeah, it's perfectly fine to depend on a newer less version if needed. But can you also propose a test that would fail if the version is too old then?

**Comment by bensadeh** [comment]
I can add a check on startup when you run the `clx` binary to see whether the installed less version is recent enough and exit with a 1 if it is not. Would this be an acceptable test?

**Reply by SMillerDev** [reply]
Yeah, that sounds okay

**Reply by bensadeh** [reply]
Thank you for confirming. I will follow up with a PR.

**Comment by bensadeh** [comment]
Hi again, I am stuck trying to write the test for the formula. 

As discussed, starting from `2.7`, the `clx` binary exits with an error code `1` if `less` is below the latest release version of `608`. 

Are you able to point me in the right direction of some resources or other formula examples to aid with formulating some good tests?

**Reply by bensadeh** [reply]
Nevermind, I think I found it.
